### ISSUE TITLE
Adding Proxy-Authorization

### ISSUE THREAD

**Issue by mwkirk** [mwkirk/javapns#104]
_Original author: i42lo...@gmail.com (February 02, 2012 08:32:29)_

If we want to use a proxy authentification, we should modify the next code in function doTunnelHandshake(Socket tunnel, String host, int port):

OutputStream out = tunnel.getOutputStream();

String msg = &quot;CONNECT &quot; + host + &quot;:&quot; + port + &quot; HTTP/1.0\n&quot; + &quot;User-Agent: BoardPad Server\n&quot;;

sun.misc.BASE64Encoder encoder = new sun.misc.BASE64Encoder();
String pwd = &quot;USER&quot; + &quot;:&quot; + &quot;PASSWORD&quot;;
String encodedUserPwd = encoder.encode(pwd.getBytes());
msg += &quot;Proxy-Authorization: &quot; + &quot;Basic &quot; + encodedUserPwd + &quot;\r\n\r\n&quot;;

Bye!

_Original issue: http://code.google.com/p/javapns/issues/detail?id=104_

**Comment by mwkirk** [comment]
_From sype...@gmail.com on February 24, 2012 04:19:49_
Fixed in r361.

Invoke ProxyManager.setProxyBasicAuthorization(username, password), or ProxyManager.setProxyAuthorization(encodedAuthorization) to configure the proxy authorization header.

**Comment by mwkirk** [comment]
_From cox1...@gmail.com on August 23, 2012 11:39:04_
Is this tested?
I just tried and got a CommunicationException:
Communication exception: java.io.IOException: Unable to tunnel through. Proxy returns "HTTP/1.1 407 Proxy Authentication Required"

I did some Java tests about how to pass the proxy. I know that it works using Apache HTTP components (HttpClient). Check their code, they have probably the right way to do it.
